src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -376,6 +376,18 @@ public function initialized($id)
     }
 
     /**
+     * Clears shared services from the container.
+     */
+    public function clear()
+    {
+        if (!empty($this->scopedServices)) {
+            throw new RuntimeException('Clearing the container is not allowed when a scope is active.');
+        }
+
+        $this->services = array();
```

Comment:
``$this->aliases`` and ``$this->methodMap`` are part of the container definition, not of the runtime usage. they are just maps of strings and so does not impact the GC of containers. This is similar for ``scopes`` and ``scopeChildren``.
And ``scopedServices`` and ``scopeStack`` will be empty here as I enforce being outside of any active scope.

Clearing it currently sets the container in the same state than just after instantiating it: if you continue to call ``get`` on it, it will just reinstantiate services (this would be equivalent to entering in a new ``container`` scope actually, which also make sense for the condition on active scopes).
Resetting all internal properties would make the code likely to break badly if someone tries to continue to use the object after clearing it.

However, the name of the method could indeed be changed.